src/components/Sidebar/Sidebar.tsx

```diff
@@ -1,115 +1,62 @@
-} from '@fortawesome/free-solid-svg-icons';
-
-import DropDown from 'components/DropDown';
-import { useTheme } from 'components/ThemeProvider';
-import { State } from 'pathways-model';
-import styles from './Sidebar.module.scss';
+import { faChevronLeft, faChevronRight, faPlus } from '@fortawesome/free-solid-svg-icons';
+
+import { SidebarHeader, SidebarButton, BranchNode, ActionNode } from '.';
+import { State, Pathway } from 'pathways-model';
+import { setStateNodeType, addTransition, createState, addState } from 'utils/builder';
+import useStyles from './styles';
 
 interface SidebarProps {
+  pathway: Pathway;
+  updatePathway: (pathway: Pathway) => void;
   headerElement: RefObject<HTMLDivElement>;
   currentNode: State;
 }
 
-interface SidebarHeaderProps {
-  currentNodeLabel: string;
-}
+const Sidebar: FC<SidebarProps> = ({ pathway, updatePathway, headerElement, currentNode }) => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(true);
+  const styles = useStyles();
+  const history = useHistory();
+  const sidebarContainerElement = useRef<HTMLDivElement>(null);
+  const currentNodeKey = currentNode?.key;
 
-const nodeTypeOptions = [
-  { label: 'Action', value: 'action' },
-  { label: 'Branch', value: 'branch' }
-];
+  const toggleSidebar = useCallback((): void => {
+    setIsExpanded(!isExpanded);
+  }, [isExpanded]);
 
-const AddNodes: FC = memo(() => {
-  return (
-    <div className={styles.addNodesContainer}>
-      <table>
-        <tbody>
-          <tr>
-            <td className={styles.button}>
-              <Button
-                variant="contained"
-                color="primary"
-                startIcon={<FontAwesomeIcon icon={faPlus} />}
-              >
-                Add Action Node
-              </Button>
-            </td>
-            <td className={styles.description}>
-              Any clinical or worfklow step which is not a decision.
-            </td>
-          </tr>
-          <tr>
-            <td className={styles.button}>
-              <Button
-                variant="contained"
-                color="primary"
-                startIcon={<FontAwesomeIcon icon={faPlus} />}
-              >
-                Add Branch Node
-              </Button>
-            </td>
-            <td className={styles.description}>
-              A logical branching point based on clinical or workflow criteria.
-            </td>
-          </tr>
-          <tr>
-            <td className={styles.button}>
-              <Button
-                variant="contained"
-                color="primary"
-                startIcon={<FontAwesomeIcon icon={faPlus} />}
-              >
-                Add Reusable Node
-              </Button>
-            </td>
-            <td className={styles.description}>
-              A previously built node or group of nodes defining a set of criteria.
-            </td>
-          </tr>
-        </tbody>
-      </table>
-    </div>
+  const changeNodeType = useCallback(
```

Comment:
Unless I am missing something it looks like all this is doing is changing the nodeType indicator and not actually changing the node type. The methods `makeBranchStateGuidance` and `makeGuidanceStateBranch` handle this

Reply:
I believe this is addressed in the latest commits.